x-pack/filebeat/tests/integration/input_reload_test.go

```diff
@@ -0,0 +1,336 @@
+					Id:   "default",
+					Type: "elasticsearch",
+					Name: "elasticsearch",
+					Source: requireNewStruct(t,
+						map[string]interface{}{
+							"type":     "elasticsearch",
+							"hosts":    []interface{}{"http://localhost:9200"},
+							"username": "admin",
+							"password": "testing",
+							"protocol": "http",
+							"enabled":  true,
+						}),
+				},
+			},
+			{
+				Id:             "input-unit-1",
+				Type:           proto.UnitType_INPUT,
+				ConfigStateIdx: 1,
+				State:          proto.State_HEALTHY,
+				LogLevel:       proto.UnitLogLevel_DEBUG,
+				Config: &proto.UnitExpectedConfig{
+					Id:   "log-input",
+					Type: "log",
+					Name: "log",
+					Streams: []*proto.Stream{
+						{
+							Id: "log-input-1",
+							Source: requireNewStruct(t, map[string]interface{}{
+								"enabled": true,
+								"type":    "log",
+								"paths":   []interface{}{logFilePath},
+							}),
+						},
+					},
+				},
+			},
+		},
+		{
+			{
+				Id:             "output-unit",
+				Type:           proto.UnitType_OUTPUT,
+				ConfigStateIdx: 1,
+				State:          proto.State_HEALTHY,
+				LogLevel:       proto.UnitLogLevel_DEBUG,
+				Config: &proto.UnitExpectedConfig{
+					Id:   "default",
+					Type: "elasticsearch",
+					Name: "elasticsearch",
+					Source: requireNewStruct(t,
+						map[string]interface{}{
+							"type":     "elasticsearch",
+							"hosts":    []interface{}{"http://localhost:9200"},
+							"username": "admin",
+							"password": "testing",
+							"protocol": "http",
+							"enabled":  true,
+						}),
+				},
+			},
+			{
+				Id:             "input-unit-2",
+				Type:           proto.UnitType_INPUT,
+				ConfigStateIdx: 1,
+				State:          proto.State_HEALTHY,
+				LogLevel:       proto.UnitLogLevel_DEBUG,
+				Config: &proto.UnitExpectedConfig{
+					Id:   "log-input",
+					Type: "log",
+					Name: "log",
+					Streams: []*proto.Stream{
+						{
+							Id: "log-input-2",
+							Source: requireNewStruct(t, map[string]interface{}{
+								"enabled": true,
+								"type":    "log",
+								"paths":   []interface{}{logFilePath},
+							}),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Once the desired state is reached (aka Filebeat finished applying
+	// the policy changes) we still wait for a little bit before sending
+	// another policy. This will allow the input to run and get some data
+	// into the publishing pipeline.
+	//
+	// nextState is a helper function that will keep cycling through both
+	// elements of the `units` slice. Once one is fully applied, we wait
+	// at least 10s then send the next one.
+	idx := 0
+	waiting := false
+	when := time.Now()
+	nextState := func() {
+		if waiting {
+			if time.Now().After(when) {
+				idx = (idx + 1) % len(units)
+				waiting = false
+				return
+			}
+			return
+		}
+		waiting = true
+		when = time.Now().Add(10 * time.Second)
+	}
+	server := &mock.StubServerV2{
+		// The Beat will call the checkin function multiple times:
```

Comment:
```suggestion
		// The Beat will call the check-in function multiple times:
```

Reply:
I'm sure I had added that to the batch...